js/src/components/paid-ads/ads-campaign/paid-ads-setup-sections.js

```diff
@@ -54,22 +61,35 @@ function resolveInitialPaidAds( paidAds ) {
  * @param {Object} props React props.
  * @param {(onStatesReceived: PaidAdsData)=>void} props.onStatesReceived Callback to receive the data for setting up paid ads when initial and also when the budget and billing are updated.
  * @param {Array<CountryCode>|undefined} props.countryCodes Country codes for the campaign.
+ * @param {Campaign} [props.campaign] Campaign data to be edited. If not provided, this component will show campaign creation UI.
+ * @param {boolean} [props.showCampaignPreviewCard=false] Whether to show the campaign preview card.
+ * @param {boolean} [props.loadCampaignFromClientSession=false] Whether to load the campaign data from the client session.
  */
 export default function PaidAdsSetupSections( {
 	onStatesReceived,
 	countryCodes,
+	campaign,
+	loadCampaignFromClientSession,
```

Comment:
I think this is a clever way of handling this. If adding a prop for controlling this ends up being undesirable, an alternate option might be to clear the clientSession data whenever the a campaign is created, and update this component to only save the clientSession data if `isCreation` is true, but I don't see this as a blocker.